extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultPKISecretEngine.java

```diff
@@ -0,0 +1,219 @@
+ */
+public interface VaultPKISecretEngine {
+
+    /**
+     * Retrieves the engine's CA certificate (PEM encoded).
+     *
+     * @return Certificate authority certificate.
+     */
+    CertificateData.PEM getCertificateAuthority();
+
+    /**
+     * Configures the engine's CA.
+     *
+     * @param pemBundle PEM encoded bundle including the CA, with optional chain, and private key.
+     */
+    void configCertificateAuthority(String pemBundle);
+
+    /**
+     * Configures engine's URLs for issuing certificates, CRL distribution points, and OCSP servers.
+     *
+     * @param options URL options.
+     */
+    void configURLs(ConfigURLsOptions options);
+
+    /**
+     * Read engine's configured URLs for issuing certificates, CRL distribution points, and OCSP servers.
+     *
+     * @return URL options.
+     */
+    ConfigURLsOptions readURLsConfig();
+
+    /**
+     * Configures engine's CRL.
+     *
+     * @param options CRL options.
+     */
+    void configCRL(ConfigCRLOptions options);
+
+    /**
+     * Read engine's CRL configuration.
+     *
+     * @return URL options.
+     */
+    ConfigCRLOptions readCRLConfig();
+
+    /**
+     * Retrieves the engine's CA chain (PEM encoded).
+     *
+     * @return Certificate authority chain.
+     */
+    String getCertificateAuthorityChain();
+
+    /**
+     * Retrieves the engine's CRL (PEM encoded).
+     *
+     * @return Certificate revocation list.
+     */
+    String getCertificateRevocationList();
+
+    /**
+     * Forces a rotation of the associated CRL.
+     */
+    boolean rotateCertificateRevocationList();
+
+    /**
+     * List all issued certificate serial numbers.
+     *
+     * @return List of certificate serialize numbers.
+     */
+    List<String> getCertificates();
+
+    /**
+     * Retrieve a specific certificate (PEM encoded).
+     *
+     * @param serial Serial number of certificate.
+     * @return Certificate or null if no certificate exists.
+     */
+    CertificateData.PEM getCertificate(String serial);
+
+    /**
+     * Generates a public/private key pair and certificate issued from the engine's CA using the
+     * provided options.
+     *
+     * @param role Name of role used to create certificate.
+     * @param options Certificate generation options.
+     * @return Generated certificate and private key.
+     */
+    GeneratedCertificate generateCertificate(String role, GenerateCertificateOptions options);
+
+    /**
+     * Generates a certificate issued from the engine's CA using the provided Certificate Signing Request and options.
+     *
+     * @param role Name of role used to create certificate.
+     * @param pemSigningRequest Certificate Signing Request (PEM encoded).
+     * @param options Certificate generation options.
+     * @return Generated certificate.
+     */
+    SignedCertificate signRequest(String role, String pemSigningRequest, GenerateCertificateOptions options);
```

Comment:
Passing in a `CertificateData.PEM` is needless complexity here. There is no option but to pass PEM formatted items to Vault. We use `CertificateData` for return values because the value can be PEM or DER and the user needs to be able to differentiate; here there is no possibility for confusion.